.github/workflows/main.yml

```diff
@@ -0,0 +1,23 @@
+name: update
+
+on:
+  workflow_dispatch:
+  repository_dispatch:
+
+jobs:
+  build:
+
+    if: github.actor == github.repository_owner
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@main
+    - name: update
+      run: bash build/update.sh
+    # Deploy the site
+    - name: Deploy
+      uses: peaceiris/actions-gh-pages@v4
```

Comment:
What advantages are there to using this custom action instead of Github's "standard" actions? I'm not familiar with either, so I was just looking at Github's suggested workflows, in this case the "static HTML" one.

```yaml
# Github's suggestion: "Static HTML"
      - name: Setup Pages
        uses: actions/configure-pages@v5
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v3
        with:
          # Upload entire repository
          path: '.'
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v4
```

Reply:
According to [the GitHub Pages documentation](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site):
> Most external CI workflows "deploy" to GitHub Pages by committing the build output to the gh-pages branch of the repository, and typically include a .nojekyll file.
> 
> Use the actions/upload-pages-artifact action to upload the static files as an artifact. 
> use the actions/deploy-pages action to deploy the artifact.

- `peaceiris/actions-gh-pages`: This action creates a gh-pages branch, and we need to configure GitHub Pages to use this `gh-pages` branch.
    - Settings -> Pages -> Build and deployment -> Source -> **Select `Deploy from a branch` -> Select `gh-pages` branch**
- `Github's "standard" actions` (`actions/upload-pages-artifact + actions/deploy-pages`): This action will create an artifact, and then we need to configure GitHub Pages to use `GitHub Actions`.
    - Settings -> Pages -> Build and deployment -> Source -> **Select `GitHub Actions`**

**I think using the gh-pages branch is better because switching branches allows you to see the generated static files.**
But the artifact needs to be downloaded to view the file, and the artifact will expire, with a default of [90 days](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#about-workflow-artifacts)(`By default, GitHub stores build logs and artifacts for 90 days, and this retention period can be customized.`)

![image](https://github.com/user-attachments/assets/489914b9-c7ef-4247-8bd9-142661c0bcd0)

![image](https://github.com/user-attachments/assets/39ddc6e4-884e-45e7-8850-7aa2810de481)

Reply:
Makes sense, thanks for explaining!